### ISSUE TITLE
bazel run with --run_under does not work with targets in external workspace

### ISSUE THREAD

**Issue by robbertvanginkel** [bazelbuild/bazel#7656]
### Description of the problem / feature request:

`--run_under` does not seem to accept targets that are defined in external workspaces:
```
$ bazel run --run_under='@externalworkspace//:wrapper' -- //:run
/bin/bash: @externalworkspace//:wrapper: No such file or directory
```

According to the `--run_under` [docs](https://docs.bazel.build/versions/master/user-manual.html#flag--run_under), this should work if the first argument to run is a fully qualified label (although it mentions "starts with `//`" as explanation for fully qualified label).

### Feature requests: what underlying problem are you trying to solve with this feature?

I'm trying to run a binary in my repo under a binary in a repo thats in another workspace. 

(Specifically I'm trying to include the go debugger delve, which is a regular go binary, using a `go_repository` rule so I can version the debugger and debug by running using the `dlv exec` command like `bazel run -c dbg --run_under='@dlv//dlv exec' -- //target`.)

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.
```
--- WORKSPACE 
local_repository(
    name = "externalworkspace",
    path = "../externalworkspace",
)
--- BUILD.bazel 
sh_binary(
  name = "run",
  srcs = ["run.sh"],
)
--- run.sh 
#!/usr/bin/env bash
echo "I ran"
--- ../externalworkspace/WORKSPACE 
local_repository(
    name = "externalworkspace",
    path = "../externalworkspace",
)
--- ../externalworkspace/BUILD.bazel 
sh_binary(
  name = "wrapper",
  srcs = ["wrapper.sh"],
)
--- ../externalworkspace/wrapper.sh 
#!/usr/bin/env bash
echo "wrapping"
$@
echo "wrapped"
```

### What operating system are you running Bazel on?

MacOS 10.14.3 (18D109)

### What's the output of `bazel info release`?
```
$ bazel info release
release 0.23.1
```
###  Have you found anything relevant by searching the web?
Did not find anything.

### Any other information, logs, or outputs that you want to share?

I tried adding an alias, as only targets starting with `//` are recognized as targets by `--run_under`. 
```
alias(
  name = "wrapper",
  actual = "@externalworkspace//:wrapper",
)
```

This doesn't result in anything helpful:
```
$ bazel run --run_under='//:wrapper' -- //:run
INFO: Analysed 2 targets (0 packages loaded, 0 targets configured).
INFO: Found 2 targets...
INFO: Elapsed time: 0.127s, Critical Path: 0.00s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
```

I tried adding `--build_event_text_file file` to debug what's going on, which triggers an `ERROR: Only a single target can be run. Do not use wildcards that match more than one target`.
```
$ bazel run --run_under='//:wrapper' --build_event_text_file file -- //:run
INFO: Analysed 2 targets (0 packages loaded, 0 targets configured).
INFO: Found 2 targets...
INFO: Elapsed time: 0.123s, Critical Path: 0.00s
INFO: 0 processes.
ERROR: Only a single target can be run. Do not use wildcards that match more than one target
INFO: Build completed successfully, 1 total action
```

**Comment by jxwufan** [comment]
Hi, I tried to hack this issue with one line modification. It seems it works, the output from the issue example is as follows:
```
➜  work ~/bazel_d/bazel-out/darwin-fastbuild/bin/src/bazel run --run_under='@externalworkspace//:wrapper' -- //:run
Starting local Bazel server and connecting to it...
INFO: Analyzed 2 targets (19 packages loaded, 157 targets configured).
INFO: Found 2 targets...
INFO: Deleting stale sandbox base /private/var/tmp/_bazel_fanwu/3fac766859cb855a15fe06cf6c04dfb4/sandbox
INFO: Elapsed time: 8.636s, Critical Path: 0.05s
INFO: 0 processes.
INFO: Build completed successfully, 4 total actions
INFO: Build completed successfully, 4 total actions
wrapping
I ran
wrapped
```

If this patch is appropriate, is there any other code like test should I write before sending a pull request?

**Comment by iirina** [comment]
@aehlig can you take a look or assign someone else if the case? Thanks!

**Comment by linzhp** [comment]
should be fixed by #8920. @robbertvanginkel could you verify?

**Comment by robbertvanginkel** [comment]
Works, thanks @jxwufan!
